cleanlab/multilabel_classification/rank.py

```diff
@@ -22,24 +22,53 @@
 
 import numpy as np  # noqa: F401: Imported for type annotations
 import numpy.typing as npt
-from typing import List, TypeVar, Dict, Any
+from typing import List, TypeVar, Dict, Any, Optional, Tuple
 
 from cleanlab.internal.validation import assert_valid_inputs
 from cleanlab.internal.util import get_num_classes
-from cleanlab.internal.multilabel_scorer import MultilabelScorer, ClassLabelScorer, Aggregator
 from cleanlab.internal.multilabel_utils import int2onehot
+from cleanlab.internal.multilabel_scorer import MultilabelScorer, ClassLabelScorer, Aggregator
 
 
 T = TypeVar("T", bound=npt.NBitBase)
 
 
+def _labels_to_binary(
+    labels: List[List[int]],
+    pred_probs: npt.NDArray["np.floating[T]"],
+) -> np.ndarray:
+    """Validate the inputs to the multilabel scorer. Also transform the labels to a binary representation."""
+    assert_valid_inputs(
+        X=None, y=labels, pred_probs=pred_probs, multi_label=True, allow_one_class=True
+    )
+    num_classes = get_num_classes(labels=labels, pred_probs=pred_probs, multi_label=True)
+    binary_labels = int2onehot(labels, K=num_classes)
+    return binary_labels
+
+
+def _create_multilabel_scorer(
+    method: str,
+    adjust_pred_probs: bool,
+    aggregator_kwargs: Optional[Dict[str, Any]] = None,
+) -> Tuple[MultilabelScorer, Dict]:
+    """This function acts as a factory that creates a MultilabelScorer."""
+    base_scorer = ClassLabelScorer.from_str(method)
+    base_scorer_kwargs = {"adjust_pred_probs": adjust_pred_probs}
+    if aggregator_kwargs:
+        aggregator = Aggregator(**aggregator_kwargs)
+        scorer = MultilabelScorer(base_scorer, aggregator)
+    else:
+        scorer = MultilabelScorer(base_scorer)
+    return scorer, base_scorer_kwargs
+
+
 def get_label_quality_scores(
     labels: List[List[int]],
     pred_probs: npt.NDArray["np.floating[T]"],
     *,
     method: str = "self_confidence",
     adjust_pred_probs: bool = False,
-    aggregator_kwargs: Dict[str, Any] = {"method": "exponential_moving_average", "alpha": 0.8}
+    aggregator_kwargs: Dict[str, Any] = {"method": "exponential_moving_average", "alpha": 0.8},
 ) -> npt.NDArray["np.floating[T]"]:
     """Computes a label quality score each example in a multi-label classification dataset.
```

Comment:
at the end of this method's docstring, add one more sentence:

This method computes an overall quality score for each example. To get a score for each class annotation for each example, use the `get_label_quality_scores_per_class` method instead.

(link that method properly in the docstring too)

Reply:
Added here: https://github.com/cleanlab/cleanlab/blob/65432bbebb1d0767fb49ae988920a2250cde3e24/cleanlab/multilabel_classification/rank.py#L109